CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/visualisation/ColorVisualisation.java

```diff
@@ -54,14 +82,46 @@ public String getName() {
     public Node getVisualisation() {
         return colorButton;
     }
+    
+    @Override 
+    public AnalyticTranslator getTranslator() {
+        return translator;
+    }
 
     @Override
     public List<SchemaAttribute> getAffectedAttributes() {
         return Arrays.asList(
-                VisualConcept.VertexAttribute.FOREGROUND_ICON,
                 VisualConcept.VertexAttribute.OVERLAY_COLOR,
                 VisualConcept.TransactionAttribute.OVERLAY_COLOR,
                 VisualConcept.GraphAttribute.NODE_COLOR_REFERENCE,
                 VisualConcept.GraphAttribute.TRANSACTION_COLOR_REFERENCE);
     }
+
+    @Override
+    public boolean isActive() {
+        return activated;
+    }
+
+    @Override
+    public void setSelected(final boolean selected) {
+        colorButton.setSelected(selected);
+        activated = selected;
+    }
+    
+    @Override 
+    public boolean equals(final Object object) {
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        } else if (this == object || getClass() == object.getClass()) {
```

Comment:
the 2nd boolean argument here will always be true if it get to this point (as the reverse argument in the if condition will be false). That will also mean that the code below the else if condition will never be called

Reply:
This is needed so that when the contains and replace checks are run on the graph visualisations map, objects of the same class are overwritten as there can only ever be one color visualisation or one size visualisation.